sample/src/main/java/org/aerogear/graphqlandroid/data/UserData.kt

```diff
@@ -108,186 +82,94 @@ class UserData(val context: Context) {
-                    Task(
-                        it, result.description(), result.id().toInt(),
-                        result.version()!!.toInt()
-                    )
-                }
-                if (obj != null) {
-                    newNotes.add(obj)
-                }
             }
-        })
-        return newNotes
-
-    }
-
-    fun OffCreateMut(mutationObj: Mutation) {
-
-        Log.e("UtilClass ", " OffCreateMut function called")
-
-        val longid = dbDao.insertMutation(mutationObj)
-
-        Log.e("UtilClass ", " OffCreateMut id of inserted mutation : $longid ")
+        }
 
-        val getInsertedMutation = dbDao.getAllMutations()
-        Log.e("UtilClass ", " OffCreateMut 2 : ${getInsertedMutation.size} ")
-        Log.e("UtilClass ", " OffCreateMut 3: ${dbDao.getAMutation(mutationObj.SNo)} ")
+        Utils.getApolloClient(context)?.enterQueue(
+            mutation as com.apollographql.apollo.api.Mutation<Operation.Data, Any, Operation.Variables>,
+            callback as ApolloCall.Callback<Any>
+        )
     }
 
     fun deleteTask(id: String) {
         Log.e(TAG, "inside delete title")
-
         val mutation = DeleteTaskMutation.builder().id(id).build()
 
-        val client = Utils.getApolloClient(context)?.mutate(
-            mutation
-        )
-        client?.enqueue(object : ApolloCall.Callback<DeleteTaskMutation.Data>() {
+        val callback = object : ApolloCall.Callback<DeleteTaskMutation.Data>() {
             override fun onFailure(e: ApolloException) {
-                Log.e("onFailure" + "deleteTask", e.toString())
-
-                val operationID = client.operation().operationId()
-                val queryDoc = client.operation().queryDocument()
-                val operationName = client.operation().name()
-                val valuemap = client.operation().variables().valueMap()
-
-                val jsonObject = JSONObject(valuemap)
-                val mutationObj = Mutation(operationID, queryDoc, operationName, jsonObject, mutation.javaClass.name)
-
-                OffDeleteMut(mutationObj)
+                Log.e("onFailure" + "updateTask", e.toString())
             }
 
             override fun onResponse(response: Response<DeleteTaskMutation.Data>) {
                 val result = response.data()?.deleteTask()
-
-                Log.e(TAG, "onResponse-DeleteTask")
-
-                Log.e(TAG, "$result")
-
-                getTasks()
+                Log.e(TAG, "onResponse-UpdateTask")
             }
-        })
-    }
-
-    fun OffDeleteMut(mutationObj: Mutation) {
-
-        Log.e("UtilClass ", " OffDeleteMut function called")
-
-        val longid = dbDao.insertMutation(mutationObj)
-
-        Log.e("UtilClass ", " OffDeleteMut id of inserted mutation : $longid ")
+        }
 
+        Utils.getApolloClient(context)?.enterQueue(
+            mutation as com.apollographql.apollo.api.Mutation<Operation.Data, Any, Operation.Variables>,
```

Comment:
Will the user need to manually cast these parameters every time?

Reply:
Yes

Reply:
I think that if you pass `UpdateCurrentTaskMutation`, that should be fine.
Every mutation extends/implements the Mutation class so similar to how the `mutate()` method in ApolloClient takes any mutation, we should be doing the same.
Forcefully casting is not only redundant but also a bad API design.

Reply:
<img width="775" alt="Screenshot 2019-07-16 at 2 41 52 PM" src="https://user-images.githubusercontent.com/33238323/61281810-e303e000-a7d7-11e9-9597-1fdce83556c4.png">

Reply:
Take a look at the `mutate()` method in apolloClient to see how they do it.

Reply:
Ok, merging this for now but we need to fix this.